routes/baseDBMethods.js

```diff
@@ -22,6 +24,35 @@ function handleDocResponse(ctx, doc) {
   }
 }
 
+function proccessDatesRangeParameters(ctx) {
+  const startDate = parseInt(ctx.query.startDate, 10) || null;
+  const endDate = parseInt(ctx.query.endDate, 10) || null;
+
+  const datesFilter = {};
+
+  if ((startDate != null) && (endDate != null) && (startDate > endDate)) {
+    throw new RangeError('Query parameter \'startDate\' can\'t be bigger than Query parameter \'endDate\'');
+  } else if ((startDate != null) || (endDate != null)) {
+    datesFilter.date = {};
+
+    if (startDate != null) {
+      if (startDate < 1) {
```

Comment:
I think the validation should be above (when you check that the start date is earlier than the end date).

Reply:
If I will do it then I will have to use if (startDate !=null) before assigning the format (with moment).
Because you need to remember at every situation we can have both NaN, both used or only one of them is.
So instead doing 3 ifs I decided to just make sure ONE of them is used and then just to check if it's startDate, if so I check for errors and assign the var, if not I check endDate.